src/Common/src/TypeSystem/Common/TargetDetails.cs

```diff
@@ -177,14 +177,50 @@ public TargetDetails(TargetArchitecture architecture, TargetOS targetOS, TargetA
             Abi = abi;
         }
 
+        /// <summary>
+        /// Gets the dyadic logarithm of the maximum size of a primitive type
+        /// </summary>
+        public static int MaximumLog2PrimitiveSize
+        {
+
+            get
+            {
+                return 3;
+            }
+        }
+
         /// <summary>
         /// Gets the maximum size of a primitive type
         /// </summary>
         public static int MaximumPrimitiveSize
         {
             get
             {
-                return 8;
+                return 1 << MaximumLog2PrimitiveSize;
+            }
+        }
+
+        /// <summary>
+        /// Return the integral value of dyadic logarithm of given size
+        /// up to MaximumLog2PrimitiveSize.
+        /// </summary>
+        /// <param name="size">Size to calculate base 2 logarithm for</param>
+        /// <returns></returns>
+        public static int GetLog2Size(int size)
```

Comment:
Could this go somewhere else? Random math functions are not exactly target details.

Reply:
I have moved this to the field layout algorithm module which is the only place where it's used.